lib/common/grpc-service.js

```diff
@@ -332,24 +364,20 @@ GrpcService.convertValue_ = function(value) {
     convertedValue = {
       structValue: GrpcService.objToStruct_(value)
     };
-  } else if (is.date(value)) {
```

Comment:
Correct me if I'm wrong, but wasn't that branch dead before? 
is.date() takes a string and this code checks for string earlier?

Going back to the timestamp topic: protoPayload cannot be used, because that requires us to know the schema upfront. 
Re [this comment ](https://github.com/GoogleCloudPlatform/gcloud-node/pull/1374#issuecomment-227601732), we have plans to make querying by timestamp strings inside json_payload easier.